test/api_tests.jl

```diff
@@ -13,24 +13,49 @@ mutable struct DiscreteVelocity <: AbstractAgent
     vel::NTuple{2, Int}
     diameter::Float64
 end
+mutable struct AmbiguousAgent <: AbstractAgent
+    id::Int
+    pos::NTuple{2, Int}
+    # The following types are abstract
+    weight::Real
+    info::AbstractString
+end
 
 @testset "Model construction" begin
     # Cannot use ImmutableAgent since it cannot be edited
     @test_throws ArgumentError ABM(ImmutableAgent)
```

Comment:
This is probably too much: we don't want to babysit the user so much as to check every single field of the agent. We should only check the following:

* The given agent type to ABM is mutable and concrete
* The given agent type's first field is `id`
* The second field `pos`'s type matches the space.

That is it, we shouldn't check for anything beyond that.

Reply:
Oh, OK. Well, if that's the case I'll remove this compatibility shim for 1.0 too.

Reply:
Have changed to a parametric type test. This also gives an example of how users who are not familiar with the type system can use inference to get started.